persist/fs/msgpack/decoder.go

```diff
@@ -139,6 +141,63 @@ func (dec *Decoder) DecodeLogEntry() (schema.LogEntry, error) {
 	return logEntry, nil
 }
 
+// DecodeLogEntryRemainingToken contains all the information that DecodeLogEntryRemaining
+// requires to continue decoding a log entry after a call to DecodeLogEntryUniqueIndex.
+type DecodeLogEntryRemainingToken struct {
+	numFieldsToSkip1 int
+	numFieldsToSkip2 int
+}
+
+// DecodeLogEntryUniqueIndex decodes a log entry as much as is required to return
+// the series unique index. Call DecodeLogEntryRemaining afterwards to decode the
+// remaining fields.
+func (dec *Decoder) DecodeLogEntryUniqueIndex() (DecodeLogEntryRemainingToken, uint64, error) {
+	if dec.err != nil {
+		return emptyLogEntryRemainingToken, 0, dec.err
+	}
+
+	numFieldsToSkip1 := dec.decodeRootObject(logEntryVersion, logEntryType)
+	numFieldsToSkip2, ok := dec.checkNumFieldsFor(logEntryType)
+	if !ok {
+		return emptyLogEntryRemainingToken, 0, errors.New("unable to determine num fields to skip")
```

Comment:
nit: move the errors.New call to a var decl

Reply:
done